tests/4.Utils.ts

```diff
@@ -34,6 +34,56 @@ describe('utils', () => {
       assert.deepStrictEqual(expected, actual);
     });
   });
+
+  describe('Base64 decoding utilities', () => {
+    it('should decode and encode Base64 roundtrip', () => {
+      const testCases = [
+        ['Hello, Algorand!', 'SGVsbG8sIEFsZ29yYW5kIQ=='],
+        ['a Ā 𐀀 文 🦄', 'YSDEgCDwkICAIOaWhyDwn6aE'],
+        ['(╯°□°）``` ┻━┻ 00\\', 'KOKVr8Kw4pahwrDvvIlgYGAg4pS74pSB4pS7IDAwXA=='],
+      ];
```

Comment:
I added this additional case, and unfortunately there was an error:

`['\x00\x01\x02\x03\x04...\xFE\xFF', 'AAECAwQuLi7+/w==']`

```
      AssertionError [ERR_ASSERTION]: Incorrect encoding of ...þÿ; got ...��
      + expected - actual

      -...��
      +...þÿ
      
      at Context.<anonymous> (tests/4.Utils.ts:48:16)
      at processImmediate (node:internal/timers:476:21)
```

Could you take a look at this?

Reply:
You're right, it does not work for non UTF-8 strings -- the following base64 string (which is different from yours) works, so I guess the encoder/decoder provides a best attempt at it:
`['\x00\x01\x02\x03\x04...\xFE\xFF', 'AAECAwQuLi7DvsO/']`

Should we use an external dependency like [base64-js](https://github.com/beatgammit/base64-js) for arbitrary binary data? It looks like we didn't do this for non UTF-8 strings to begin with.

Reply:
If I use the exact conversion functions from https://developer.mozilla.org/en-US/docs/Glossary/Base64#the_unicode_problem (copied below), I find it is able to properly encode and decode the base64 string `AAECAwQuLi7+/w==` to the Uint8Array `[0, 1, 2, 3, 4, 46, 46, 46, 254, 255]`.

So I am confident base64 <=> Uint8Array conversion can work with this method.

Maybe something's going wrong with the Uint8Array <=> string conversion? I will try to look at that more closely soon.

```javascript
function base64ToBytes(base64) {
  const binString = atob(base64);
  return Uint8Array.from(binString, (m) => m.codePointAt(0));
}

function bytesToBase64(bytes) {
  const binString = Array.from(bytes, (x) => String.fromCodePoint(x)).join("");
  return btoa(binString);
}
```

If you find this implementation works equally well

Reply:
Yes, I used the same implementation as that one -- the problem only exists when trying to go from Uint8Array with non UTF-8 characters to a (UTF-8) string and back into something else.

If we're satisfied with just checking that the base64 (with non UTF-8 bytes) can go to binary data, then we're good and I can update the tests for this case.

Reply:
I added a separate test case for this in the latest commit as an example: https://github.com/algorand/js-algorand-sdk/pull/800/files#diff-9edc36253646a135dddbc24bad91040c7e13cda76d51a79e2c8737d082a85de9R65

Reply:
After more research, I've concluded that since Javascript strings are limited to UTF-8, it makes sense that they cannot hold those invalid UTF-8 characters (e.g. `\xff`).

As long as the new `TextDecoder` code path produces the same output as the `Buffer` codepath does for these invalid characters, this seems like an ok result to me. Could you add a test that checks this?

Reply:
Thanks for the double check, I added a test for this here (with only UTF-8 characters): https://github.com/algorand/js-algorand-sdk/pull/800/files#diff-03788da9bcb72be6b53a86f12c45248552ca8732d28938fc18d6bdebdbf64ccfR604

It should run both code paths (`Buffer` in node and `TextDecoder` in browser) in CI.

Reply:
I added `'/v8='` to the roundtrip test case that checks for `base64ToString` conversions - let me know if it resolves this: [92333c2](https://github.com/algorand/js-algorand-sdk/pull/800/commits/92333c25147cc55a3ca15c0bfffd714c77d48b93)